mbed-hal/spi_api.h

```diff
@@ -90,17 +139,26 @@ void spi_free(spi_t *obj);
  * @param[in]     bits  The number of bits per frame
  * @param[in]     mode  The SPI mode (clock polarity, phase, and shift direction)
  * @param[in]     order Bit order. SPI_MSB (standard) or SPI_LSB.
+ *
+ * @return  rc  `SPI_RC_OK` or
+ *              `SPI_RC_UNSUPPORTED_NUMBER_OF_BITS` or
+ *              `SPI_RC_UNSUPPORTED_MODE` or
+ *              `SPI_RC_UNSUPPORTED_BIT_ORDER`
```

Comment:
we should enumarate return values with \retval <return value> { description } (Note to myself: we should write RFC for coding style, to be concise in code and documentation :) )